components/builder-jobsrv/src/server/worker_manager.rs

```diff
@@ -197,7 +203,13 @@ impl WorkerMgr {
                         continue;
                     }
                 }
-                None => break,
+                None => {
+                    debug!("no workers available - bailing for now");
+                    job.set_state(jobsrv::JobState::Pending);
+                    self.datastore.set_job_state(&job)?;
```

Comment:
Will this insert a row into postgres? Is it possible this could flood the db during inactivity with every cycle of the loop inserting rows?